### ISSUE TITLE
Database Performance Opportunity

### ISSUE THREAD

**Issue by larsborn** [Malshare/Frontend#4]
We are sending quite some queries like
```
(SELECT id FROM tbl_sample_sources WHERE source like '%somethingsomething%' LIMIT 1000)
```
to the database. Using wildcards both in the beginning _and_ the end is hard to optimize for MySQL.

**Comment by silascutler** [comment]
Yeah, I'll start looking into ways to improve this.  When it was initially written I wanted to ensure support for searching for parts of the URI path - Ex:

![image](https://user-images.githubusercontent.com/2154372/65521948-01f9af00-deb8-11e9-921e-24dcd9a6b37c.png)

**Comment by larsborn** [comment]
On September 24, 2019 11:43:10 PM GMT+09:00, Silas Cutler <notifications@github.com> wrote:
>Yeah, I'll start looking into ways to improve this.  When it was
>initially written I wanted to ensure support for searching for parts of
>the URI path - Ex:
>
>![image](https://user-images.githubusercontent.com/2154372/65521948-01f9af00-deb8-11e9-921e-24dcd9a6b37c.png)

I think a FULLTEXT index would be the way to go then.

Of you like, I can take care of this in a few weeks!
-- 
Festnetz: +49 228 304 137 03
Mobil: +49 178 689 0164
eMail: lars@wallenborn.net
Webseite: www.wallenborn.net

**Comment by silascutler** [comment]
Sounds good.  I'll leave it to you
